### ISSUE TITLE
Password with special characters doesn´t work in Fish/Bash but works in Powershell

### ISSUE THREAD

**Issue by childoftheabyss** [restic/restic#4642]
Output of `restic version`
--------------------------
restic 0.16.2 compiled with go1.21.3 on linux/amd64

What backend/service did you use to store the repository?
---------------------------------------------------------
S3 (iDrive Cloud) 

Problem description / Steps to reproduce
----------------------------------------

Hi,
I configured a repository in a Windows machine using a Powershell script and successfully backup some data to a S3 backend (I could list the snapshots, etc). Then, later, I tried to access the same repository via a Linux machine, using the same credentials and password I set up before. I ran the following command in Fish terminal:

`restic -r s3:<S3_URL> snapshots`

However I got the error `Fatal: wrong password or no key found`. I re-ran using the flag `--verbose=2` and the output was the same.

I checked the (very long) password and realized it contains many special characters, like `@`, backticks, `\`, etc. So I single-quoted the `RESTIC_PASSWORD` env var and the error was the same. I also checked the exported S3 credentials, and they were correct. So, I tried ran the same restic command in a Bash terminal, the output did not change. I tried to escape the special characters in many ways, with no avail.

So, I tried to use the the same Powershell script in my Linux machine by invoking `pwsh script.ps1` and it was able to successfully list all my snapshots. The access key, secret key and password were exactly the same. In one more test, I tried to use the `--password-file` flag in Fish, and pasting the password in the file as is. The output was the fatal error above. So, it looks like restic behaves differently in respect to special characters in Windows/Powershell and Linux/Bash/Fish.

As another check, I created another repository using a password with no special characters and same S3 credentials using Powershell and I was able to successfully list snapshots via Fish.

After typing this issue, I had the idea of cloning the restic repository to my machine and tried to debug via VS Code. I discovered that, for example, in `SearchKey` function in `key.go` the password variable is printed correctly when I do  `fmt.Println(password)`, however in the watch tab the variable is shown with the backslashs duplicated, i.e. `\\`. I am almost sure this was due my need of escaping the backslashs while passing to VS Code `launch.json` configuration, but would not Go understand the double backslash as escape and the variable would have only one backslash? I tried hardcoding the password `pwd` in `resolvePassword` method in `global.go` (because looked like where it gets from the environment) and the result is the same: duplicated backslashes in watch tab but correct `fmt.Print()` output. Well, I am not a programmer but I know some basic code, and this is how far I went into troubleshooting. I don´t have a Windows machine with me to test and check the difference.

Final part of debug log:

```
2024/01/20 01:31:16 logger/log.go:47	logger.(*Backend).Load	1	  load err <nil>
2024/01/20 01:31:17 repository/key.go:148	repository.SearchKey.func1	1	key <key_here> returned error ciphertext verification failed
2024/01/20 01:31:17 logger/log.go:61	logger.(*Backend).List	1	  list err <nil>
2024/01/20 01:31:17 restic/cleanup.go:87	main.Exit	1	exiting with status code 1
```

Expected behavior
-----------------

restic should behave well when password has special characters, specially when using a password file.

Actual behavior
---------------

In Powershell, environment variable `RESTIC_PASSWORD` can contain special characters. In Bash and Fish, throws `Fatal: wrong password or no key found`.

Do you have any idea what may have caused this?
-----------------------------------------------

Maybe some internal escaping failing? Looks like Golang expects double backslashes to escape backslash, however it prints one thing and seems it save as variable another thing. I read somewhere about using backticks to escape such string, however my password also contain backticks. I don´t know why Powershell behaves correctly.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
It did! It is easy to use, well-documented, the snapshot feature is great and gives me some peace knowing I can use to securely backup my data.

**Comment by MichaelEischer** [comment]
The password problems you have are either a problem with quoting on the bash/fish or powershell side. Restic literally uses the password it gets in the environment variable without any internal quoting.

> In one more test, I tried to use the `--password-file` flag in Fish, and pasting the password in the file as is. The output was the fatal error above.

That means that the quoting on the powershell side is wrong and powershell unintentionally interprets something in that password. The password file must contain the exact bytes that powershell passes to restic.

**Comment by childoftheabyss** [comment]
Hi,
thank you for replying. I investigated further today and finally discovered the problem. It not restic fault at all, but Powershell, I did not know the `` ` `` (backtick) was the escape character there, so I had problem with some characters in password due to this that was interpreted not literally, but "creatively" (I think I have used a environment variable instead of pasting the password while creating the repository). As you said, after I made Fish to understand the same bytes by manually converting the final "interpreted" string, I was able to access the repository.

For those in similar situation, echo the password variable in Powershell, the output will be how it understand the string.

Thank you for your time. restic is a great tool and works smoothly for me. I am closing this issue.
